### PR TITLE
Add more in-depth tests of the TLS stack

### PR DESCRIPTION
**Pull request by jaredly**
jit-tests and interpreter tests should pass!

### CODE REVIEW

File: scheme-libs/racket/unison/tls.rkt [hunk 1]
```diff
@@ -89,6 +95,9 @@
        [(lambda err
           (string-contains? (exn->string err) "not valid for hostname"))
         (lambda (e) (exception "IOFailure" "NameMismatch" '()))]
+       [(lambda err
+          (string-contains? (exn->string err) "certificate verify failed"))
+        (lambda (e) (exception "IOFailurez" "certificate verify failed" '()))]
```
Comment (dolio) [inline scheme-libs/racket/unison/tls.rkt]: Is 'Failurez' a typo?

**Review by dolio — APPROVED** [review]
Looks good, I think, aside from the possible typo.

**Review by pchiusano — APPROVED** [review]
Nice!!

I'll wait for @SystemFw review before merging, and we may want to merge #3905 first? (Or merge this into #3905 ?)

@SystemFw let us know what you'd like to do.

**Comment by SystemFw** [comment]
I would probably merge this one to main as soon as the typo comment is addressed, then do the rebasing on mine before merging that

**Comment by jaredly** [comment]
done!

**Pull request merged**